lua/nui/popup/init.lua

```diff
@@ -63,28 +63,49 @@ local function calculate_window_size(size, container)
   }
 end
 
+local function apply_pin(dir, container, size)
+if dir == 'bot' or dir == 'bottom' then dir = 'btm' end
+
+  local pin    =  {
+    top        =  0,
+    right      =  container.size.width - size.width,
+    btm        =  container.size.height - size.height - 3,
+    left       =  0,
+    }
+
+  return pin[dir]
+end
+
+
 local function calculate_window_position(position, size, container)
   local row
   local col
 
   local is_percentage_allowed = not vim.tbl_contains({ "buf", "cursor" }, container.relative)
+  local is_pin_allowed = not vim.tbl_contains({"cursor"}, container.relative)
```

Comment:
Just realized I didn't add corresponding assert() in the is_pin if statements below...